File: madameautomata-bot-python-autoreg/bot/commands/punish.py

import discord
from discord import app_commands, Interaction
from ..theme import gothic_embed
from ..util.perm import is_domme, is_staff, is_sub
from ..db import Session, User, Collaring, Cage, JailSession, Star, Task, Worship, Config, CollarType, StarStatus
from ..util.time import parse_duration, random_case
from ..util.ids import cuid
from datetime import datetime, timedelta
from sqlalchemy import select, func, update, and_

def setup(bot):
    @bot.tree.command(name="punish", description="Issue a punishment to a sub.")
    @app_commands.describe(sub="Submissive", reason="Punishment details")
    async def punish(i: Interaction, sub: discord.User, reason: str):
        member = i.guild.get_member(i.user.id)
        if not await is_domme(member):
            return await i.response.send_message("Only Dommes can use this.", ephemeral=True)
        await i.response.send_message(embed=gothic_embed("Punishment", f"**<@{sub.id}>** You are punished: {reason}"))

File: madameautomata-js-bot/src/commands/punish.js

import { SlashCommandBuilder, PermissionFlagsBits, ChannelType, ActionRowBuilder, ButtonBuilder, ButtonStyle, StringSelectMenuBuilder, EmbedBuilder } from "discord.js";
import { prisma } from "../lib/prisma.js";
import { Theme, gothicEmbed } from "../theme/theme.js";
import { parseDuration, randomCase } from "../util/duration.js";
import { cuid } from "../util/ids.js";
import { isDomme, isSub, isStaff, getConfig } from "../util/perm.js";

const cmd = 
{
  data: new SlashCommandBuilder()
    .setName("punish")
    .setDescription("Issue a punishment to a sub.")
    .addUserOption(o=>o.setName("sub").setDescription("Submissive").setRequired(true))
    .addStringOption(o=>o.setName("reason").setDescription("Punishment details").setRequired(true)),
  async run(i) {
    if (!await isDomme(i.member)) return i.reply({ content: "Only Dommes can use this.", ephemeral: true });
    const sub = i.options.getUser("sub", true);
    const reason = i.options.getString("reason", true);
    return i.reply({ embeds: [gothicEmbed("Punishment", `**<@${sub.id}>** You are punished: ${reason}`)] });
  }
}


export default cmd;
